src/meta/server_state.cpp

```diff
@@ -2943,6 +2944,117 @@ bool server_state::validate_target_max_replica_count(int32_t max_replica_count)
     return valid;
 }
 
+void server_state::on_start_manual_compact(start_manual_compact_rpc rpc)
```

Comment:
The function is too long

Reply:
I agree with you, this function will do following things:
1. check app validation
2. parse request parameter into manual compaction app envs
3. update app envs
4. reset local status

What's your suggestion about how to make think function shorter?

Reply:
split this function into four small functions, according to your comment above

Reply:
I add a new function `update_compaction_envs_on_remote_storage` to implement the step2 and step3, remain step1 and step4 in original `on_start_manual_compact` function.